### PR TITLE
Specified C++17 version for CMake when installing on Windows. 

### PR DESCRIPTION
**Pull request by danieleboaretti**
Specified C++17 version for CMake when installing on Windows. I tested this change locally and it solves the error reported here https://github.com/NOhs/affine_transform_nd/issues/27#issuecomment-1519114678

### CODE REVIEW

**Comment by nogakertes** [comment]
Hi, How can I specify C++17 version for CMake?

**Comment by NOhs** [comment]
I will merge it as soon as I have a Windows test runner up and running again. Thanks for the PR!

**Comment by NOhs** [comment]
Please rebase to the lastest master and check that all Windows tests for this PR are green, thank you 🙂

**Comment by danieleboaretti** [comment]
Thanks a lot! The tests are green if I look at https://github.com/danieleboaretti/affine_transform_nd/actions/runs/4965799388  😄 
After merging this PR, for me you can close https://github.com/NOhs/affine_transform_nd/issues/27

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ add_subdirectory(extern/pybind11)
 if(MSVC)
```
Comment (NOhs) [inline CMakeLists.txt:10]:
```cmake
set(CMAKE_CXX_STANDARD 17)

if(MSVC)
    add_compile_options("/W4" "/Ox" "/fp:fast")
endif()
if(UNIX)
    add_compile_options("-Ofast")
endif()
```

I think we can simplify here. Could you try this out?

Comment (NOhs) [reply CMakeLists.txt:10]: Something is off, why is only codacy static code analysis running? 🤔

Comment (danieleboaretti) [reply CMakeLists.txt:10]: The tests are also running in the workflows as far as I can see here https://github.com/danieleboaretti/affine_transform_nd/actions (the publish workflows fail due to missing rights and that makes sense to me 😄 )

The tests pass with the simplification you suggested! 🙌

Comment (NOhs) [reply CMakeLists.txt:10]: I see, so they are triggered in your fork not this main repository. Kind of difficult to keep track of. I will have to see if I can play around with the settings. Thanks for the hint.

**Review by NOhs — APPROVED** [review]

**Pull request merged**